sktime/forecasting/hf_transformers_forecaster.py

```diff
@@ -315,16 +382,45 @@ def _predict(self, fh, X=None):
         fh = fh.to_relative(self.cutoff)
 
         self.model.eval()
-        from torch import from_numpy
+        hist_y = y if self._global_forecasting else self._y
+
+        if not isinstance(hist_y.index, pd.MultiIndex):
+            hist_y = _to_multiindex(hist_y)
+            if X is not None:
+                X = _to_multiindex(X)
+                _X = _to_multiindex(self._X)
+            converted_to_multiindex = True
+        else:
+            _X = self._X
+            converted_to_multiindex = False
+
+        hist = _frame2numpy(hist_y).squeeze(2)
 
-        hist = self._y.values.reshape((1, -1))
         if X is not None:
-            hist_x = self._X.values.reshape((1, -1, self._X.shape[-1]))
-            x_ = X.values.reshape((1, -1, self._X.shape[-1]))
+            if not self._global_forecasting:
+                hist_x = _frame2numpy(_X)
+                x_ = _frame2numpy(X)
+            else:
+                len_levels = len_levels = len(X.index.names)
+                hist_x = _frame2numpy(
```

Comment:
Can you add a comment here, what this statement with the groupby is doing? 
This is quiet hard to read and without any comments it will be hard to maintain.